crates/cli/cli-components/src/impls/commands/queries/connection_end.rs

```diff
@@ -0,0 +1,89 @@
+use core::marker::PhantomData;
+
+use cgp_core::prelude::*;
+use hermes_relayer_components::build::traits::builders::chain_builder::CanBuildChain;
+use hermes_relayer_components::chain::traits::queries::chain_status::CanQueryChainHeight;
+use hermes_relayer_components::chain::traits::queries::connection_end::CanQueryConnectionEnd;
+use hermes_relayer_components::chain::traits::types::chain_id::HasChainId;
+use hermes_relayer_components::chain::traits::types::connection::HasConnectionEndType;
+use hermes_relayer_components::chain::traits::types::height::HasHeightType;
+use hermes_relayer_components::multi::traits::chain_at::HasChainTypeAt;
+use hermes_relayer_components::multi::types::index::Index;
+
+use crate::traits::build::CanLoadBuilder;
+use crate::traits::command::CommandRunner;
+use crate::traits::output::CanProduceOutput;
+use crate::traits::parse::CanParseArg;
+
+pub struct RunQueryConnectionEndCommand;
+
+#[derive(Debug, clap::Parser, HasField)]
+pub struct QueryConnectionEndArgs {
+    #[clap(
+        long = "chain",
+        required = true,
+        value_name = "CHAIN_ID",
+        help_heading = "REQUIRED",
+        help = "Identifier of the chain to query"
+    )]
+    chain_id: String,
+
+    #[clap(
+        long = "connection",
+        visible_alias = "conn",
+        required = true,
+        value_name = "CONNECTION_ID",
+        help_heading = "REQUIRED",
+        help = "Identifier of the connection to query"
+    )]
+    connection_id: String,
+
+    #[clap(
+        long = "height",
+        value_name = "HEIGHT",
+        help = "Height of the state to query. Leave unspecified for latest height."
+    )]
+    height: Option<String>,
+}
+
+impl<App, Args, Build, Chain, Counterparty> CommandRunner<App, Args>
+    for RunQueryConnectionEndCommand
+where
+    App: CanLoadBuilder<Builder = Build>
+        + CanProduceOutput<Counterparty::ConnectionEnd>
```

Comment:
The connection end returned from `CanQueryConnectionEnd` is of type `Chain::ConnectionEnd`. So you need to use that to produce the CLI output.